robottelo/host_helpers/capsule_mixins.py

```diff
@@ -38,3 +41,37 @@ def wait_for_tasks(
         else:
             raise AssertionError(f"No task was found using query '{search_query}'")
         return tasks
+
+    def wait_for_sync(self, timeout=600, start_time=datetime.utcnow()):
+        """Wait for capsule sync to finish and assert the sync task succeeded"""
+        # Assert that a task to sync lifecycle environment to the capsule
+        # is started (or finished already)
+        logger.info(f"Waiting for capsule {self.hostname} sync to finish ...")
+        sync_status = self.nailgun_capsule.content_get_sync()
+        logger.info(f"Active tasks {sync_status['active_sync_tasks']}")
+        assert (
+            len(sync_status['active_sync_tasks'])
+            or datetime.strptime(sync_status['last_sync_time'], '%Y-%m-%d %H:%M:%S UTC')
+            > start_time
```

Comment:
I am not the author of this function, I have just moved it from a test module to `capsule_mixins.py`. It was introduced in #9629, so let me redirect your question to the function author.
@vsedmik do you have any thoughts about this one?

Reply:
The meaning was to assert the sync task is either still running, or it was finished already (e.g. for small repos when it was triggered on a CV promotion) and since we now share one capsule for all tests in the module, we wanted to ensure the `last_sync_time` does not come from previous test run.

I just tried locally to remove the datetime part and it broke 3 tests, so obviously the part was True 3 times!
Though I can see Jitendra's point that the correct way would be pass the time acquired just after CV promotion as parameter, I would vote to keep it as is, or to remove the first assert completely as we verify the content is synced on the capsule anyway (meaning the sync WAS run).

Reply:
I suggest to move these comments to JIRA card for future upgrade scenario tests refactor as this seems out of scope of this PR @jyejare